src/linalg.jl

```diff
@@ -208,17 +208,24 @@ end
 
 #--------------------------------------------------
 # Norms
-@inline LinearAlgebra.norm_sqr(v::StaticVector) = mapreduce(abs2, +, v; init=zero(real(eltype(v))))
+function _inner_eltype(v::AbstractArray)
+    return eltype(v) <: AbstractArray ? _inner_eltype(first(v)) : eltype(v)
+end
+_inner_eltype(x::Number) = typeof(x)
+@inline _init_zero(v::AbstractArray) = zero(float(real(_inner_eltype(v))))
+@inline function LinearAlgebra.norm_sqr(v::StaticArray)
+    return mapreduce(LinearAlgebra.norm_sqr, +, v; init=_init_zero(v))
```

Comment:
```suggestion
    return sum(LinearAlgebra.norm_sqr, v)
```
How about doing it like that? Would there be some drawback?

Reply:
Hm, perhaps the zero-initialization would diverge between this branch and other below... and the zero-initialization of `sum` doesn't fit here.